types/src/v2/params.rs

```diff
@@ -0,0 +1,193 @@
+	where
+		E: de::Error,
+	{
+		match s {
+			"2.0" => Ok(TwoPointZero),
+			_ => Err(de::Error::invalid_value(Unexpected::Str(s), &self)),
+		}
+	}
+}
+
+impl<'de> Deserialize<'de> for TwoPointZero {
+	fn deserialize<D>(deserializer: D) -> Result<TwoPointZero, D::Error>
+	where
+		D: Deserializer<'de>,
+	{
+		deserializer.deserialize_str(TwoPointZeroVisitor)
+	}
+}
+
+impl Serialize for TwoPointZero {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: Serializer,
+	{
+		serializer.serialize_str("2.0")
+	}
+}
+
+/// Parameters sent with the RPC request
+#[derive(Clone, Copy, Debug)]
+pub struct RpcParams<'a>(Option<&'a str>);
+
+impl<'a> RpcParams<'a> {
+	/// Create params
+	pub fn new(raw: Option<&'a str>) -> Self {
+		Self(raw)
+	}
+
+	/// Attempt to parse all parameters as array or map into type T
+	pub fn parse<T>(self) -> Result<T, RpcError>
+	where
+		T: Deserialize<'a>,
+	{
+		match self.0 {
+			None => Err(RpcError::InvalidParams),
+			Some(params) => serde_json::from_str(params).map_err(|_| RpcError::InvalidParams),
+		}
+	}
+
+	/// Attempt to parse only the first parameter from an array into type T
+	pub fn one<T>(self) -> Result<T, RpcError>
+	where
+		T: Deserialize<'a>,
+	{
+		self.parse::<[T; 1]>().map(|[res]| res)
+	}
+}
+
+/// [Serializable JSON-RPC parameters](https://www.jsonrpc.org/specification#parameter_structures)
+///
+/// If your type implement `Into<JsonValue>` call that favor of `serde_json::to:value` to
+/// construct the parameters. Because `serde_json::to_value` serializes the type which
+/// allocates whereas `Into<JsonValue>` doesn't in most cases.
+#[derive(Serialize, Debug)]
+#[serde(untagged)]
+pub enum JsonRpcParams<'a> {
+	/// No params.
+	NoParams,
+	/// Positional params (heap allocated)
+	Array(Vec<JsonValue>),
+	/// Positional params (slice)
+	ArrayRef(&'a [JsonValue]),
+	/// Params by name.
+	Map(BTreeMap<&'a str, JsonValue>),
+}
+
+impl<'a> From<BTreeMap<&'a str, JsonValue>> for JsonRpcParams<'a> {
+	fn from(map: BTreeMap<&'a str, JsonValue>) -> Self {
+		Self::Map(map)
+	}
+}
+
+impl<'a> From<Vec<JsonValue>> for JsonRpcParams<'a> {
+	fn from(arr: Vec<JsonValue>) -> Self {
+		Self::Array(arr)
+	}
+}
+
+impl<'a> From<&'a [JsonValue]> for JsonRpcParams<'a> {
+	fn from(slice: &'a [JsonValue]) -> Self {
+		Self::ArrayRef(slice)
+	}
+}
+
+/// Id of a subscription, communicated by the server.
+#[derive(Debug, PartialEq, Clone, Hash, Eq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+#[serde(untagged)]
+pub enum SubscriptionId {
+	/// Numeric id
+	Num(u64),
+	/// String id
+	Str(String),
+}
+
+impl From<SubscriptionId> for JsonValue {
+	fn from(sub_id: SubscriptionId) -> Self {
+		match sub_id {
+			SubscriptionId::Num(n) => n.into(),
+			SubscriptionId::Str(s) => s.into(),
+		}
+	}
+}
+
+/// Request Id
+#[derive(Debug, PartialEq, Clone, Hash, Eq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+#[serde(untagged)]
+pub enum Id {
```

Comment:
decided to bring this back, currently we just use `Number` is the clients and the servers is using RawValue

but probably worth to comply with the spec